common/src/main/java/com/wynntils/handlers/actionbar/ActionBarHandler.java

```diff
@@ -20,8 +20,8 @@
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public final class ActionBarHandler extends Handler {
-    // example: "§c❤ 218/218§0    §7502§f S§7 -1580    §b✺ 1/119"
-    private static final Pattern ACTIONBAR_PATTERN = Pattern.compile("(?<LEFT>§[^§]+)(?<CENTER>.*)(?<RIGHT>§[^§]+)");
+    // https://regexr.com/7kfrm
+    private static final Pattern ACTIONBAR_PATTERN = Pattern.compile("(.+?) {4,}(.+?) {4,}(.+)");
```

Comment:
What does `.+?` mean? It looks like invalid regexp syntax to me. I assume you mean just `.+`, that is, one or more of any character. If you mean zero or more, use `.*`.

I'm also thinking that someone cared enough to annotate the groups with names in the old pattern. (Maybe it was me, but I don't recall doing it). Even if it is not used, someone felt it was helpful, so maybe keep that?

Reply:
? can be added as an additional modifier to */+. It will match the **least** amount or characters possible.

Reply:
Another thing that always makes me think twice in regexps like this is how the greediness works. I usually get it wrong, so I prefer coding safely. If you have this regexp `(.+) {4,}(.+)` and then you match the string "abc     def", where there is five spaces, will it match as (abc)(     )(def) with five spaces initially or (abc)(    )( abc) with four spaces initially? The answer depends on how you set the greediness behavior of the regexp engine.

To avoid this, you can instead code it like `(.+) {4,}([^ ].*)`, thus forcing the last part to never under any circumstances start with a space. While the regexp gets longer, I tend to prefer this kind of writing, since it eliminates any question about how stuff will match in "ambiguous" inputs.

(In your case, the corresponding change would also need to be done for the "end" of the middle segment)

Reply:
`.+?` is lazy matching, if we don't use lazy, the first matcher tries to steal spaces and only leave the ` {4,}` four spaces. Same goes for the middle one. Without it, the cg's sometimes ends up with extra trailing spaces. Last one does not need to be lazy since there are no spaces after the right segment

Reply:
Ah, I see. Well, this solves the problem I was worried about. Turns out you can still learn new stuff about regexps after 25 years. ;-)